CMakeLists.txt

```diff
@@ -898,12 +902,6 @@ include (CTest)
 #
 # To disable -Werror, pass -Wno-error to Seastar_CXX_FLAGS.
 #
-# We disable _FORTIFY_SOURCE because it generates false positives with longjmp() (src/core/thread.cc)
-#
-
-target_compile_options (seastar
```

Comment:
To educate myself.
Before the patch FORTIFY_SOURCE was off for the whole seastar. Now it's off for thread.cc only, is that correct?
If so, do we expect any performance penalties once it's enabled for most of the seastar?

Reply:
> To educate myself. Before the patch FORTIFY_SOURCE was off for the whole seastar. 

yes, it's undefined for seastar, and also for the projects which use seastar via its .pc file and those which use seastar via its CMake config files.

> Now it's off for thread.cc only, is that correct?

correct.

> If so, do we expect any performance penalties once it's enabled for most of the seastar?

i think so. as the fortified version of glibc function do perform extra checks. i think, the next question would be "how much?". as an application can always override the level of the fortification level at the compile time using something like `-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3`. so the answer might depend on what the compile options are used.

if the overhead is not negligible, or we don't have a benchmark to justify the change, i think the safe move is probably to 

1. keep this `-U_FORTIFY_SOURCE` private and apply it only to `src/core/thread.cc` as in this commit
2.  add this option back globally in Seastar as a private option when compiled in Debug or Sanitize modes.

Reply:
sorry, i haven't read the 2nd commit. so it has been implemented. thanks!

Reply:
2nd commit does this

```patch
+set(FORTIFY_BUILD_TYPES "Release;RelWithDebInfo;Dev")
+if(CMAKE_BUILD_TYPE IN_LIST FORTIFY_BUILD_TYPES)
+  add_definitions(-D_FORTIFY_SOURCE=2)
+endif()
+
```

I see the final result as -- set FORTIFY_SOURCE=2 for Release build. Is level 2 considered to be "no penalty"?

Reply:
no, i don't think so. it incurs less penalty than level 3. to be on the safe side, i think we should undefine this macro for `Release` and `RelWithDebInfo`. as to Dev, we can enable level 2 or even level 3.

Reply:
I agree Seastar should leave it to the application and not change defaults.

Can we just add #undef (plus comment explaining why) to the source file?